pinot-core/src/main/java/com/linkedin/pinot/core/common/BlockSingleValIterator.java

```diff
@@ -21,30 +21,31 @@
  */
 public abstract class BlockSingleValIterator implements BlockValIterator {
 
-
-  char nextCharVal(){
+  char nextCharVal() {
     throw new UnsupportedOperationException();
   }
 
-  public int nextIntVal(){
+  public int nextIntVal() {
     throw new UnsupportedOperationException();
   }
 
-  public float nextFloatVal(){
+  public float nextFloatVal() {
     throw new UnsupportedOperationException();
   }
 
-  public long nextLongVal(){
+  public long nextLongVal() {
     throw new UnsupportedOperationException();
   }
 
-  public double nextDoubleVal(){
+  public double nextDoubleVal() {
     throw new UnsupportedOperationException();
   }
 
-  public byte[] nextBytesVal(){
+  public byte[] nextBytesVal() {
     throw new UnsupportedOperationException();
   }
 
-
+  public String nextStringVal() {
```

Comment:
This one is never called, and currently we don't use BlockValIterator to get values. Currently we use BlockValIterator to get dictionary ids, which is very confusing. Add a TODO and we should only use BlockValIterator to get values but not dictionary ids. For dictionary id, maybe we can add a BlockDictIdIterator.